### ISSUE TITLE
IPv6 question

### ISSUE THREAD

**Issue by nextgeofront** [eclipse/leshan#661]
Hi ~ there.

I have  wondering things that leshan-server-demo is supported IPv6 properly.

On IPv6 environments, I think that leshan-client-demo can't receive registered message sent from leshan-server-demo. so leshan-server-demo de-register the client registrered after 30 secs.

Between my client PC(leshan-client-demo) and server PC(leshan-server-demo), ping6 command is okay.

But, On IPv4 environments, Everything is okay.

I need some advices about this problem of IPv6.

**Comment by sbernard31** [comment]
> I think that leshan-client-demo can't receive registered message sent from leshan-server-demo. so leshan-server-demo de-register the client registrered after 30 secs.

What make you think that ?

[Here some hint](https://github.com/eclipse/leshan/issues/636#issuecomment-453528719) about ipv6 and Leshan.

**Comment by nextgeofront** [comment]
Thanks your advice. 

I already tried to use  -Djava.net.preferIPv6Addresses=true and direct ipv6 address on client side.
but client couldn't receive registered message from server. 

our ipv6 addresses are private ipv6 addresses for testing. is that problem?

**Comment by sbernard31** [comment]
Hard to say ...but I would bet this is more a network issue than a Leshan one.
Let us know if you find any solution.

**Comment by nextgeofront** [comment]
I solved my problem. I made a mistake. 

I used the first ipv6 ipaddress for server :
=====
inet6 2001:2:0:aab1::1 prefixlen 64 <--- used
inet6 2001:2::aab1:cb4:eefa:46c5:2ab3 prefixlen 64 autoconf secured
inet6 2001:2::aab1:55f:da91:15eb:b504 prefixlen 64 autoconf temporary
=====

When i use third ipv6 address, every communication is okay.
====
inet6 2001:2:0:aab1::1 prefixlen 64 
inet6 2001:2::aab1:cb4:eefa:46c5:2ab3 prefixlen 64 autoconf secured
inet6 2001:2::aab1:55f:da91:15eb:b504 prefixlen 64 autoconf temporary <-- used
====

Thank you for your network advice. 
I appreciate it.
